drivers/modem/gsm_ppp.c

```diff
@@ -1174,6 +1205,10 @@ static int gsm_init(const struct device *dev)
 		return -ENODEV;
 	}
 
+	net_mgmt_init_event_callback(&gsm_mgmt_cb, gsm_mgmt_event_handler,
+				     NET_EVENT_IF_DOWN);
```

Comment:
If you only listen for `NET_EVENT_IF_DOWN` here, there's no point to handle for `NET_EVENT_IF_UP` in the handler.

Reply:
Thanks for the feedback, I processed your review comments. Could you validate that this fix in combination #41802 works in your case as well?